core/domain/state_domain_test.py

```diff
@@ -64,43 +109,203 @@ def test_get_all_html_content_strings(self):
+        state2.update_interaction_id('DragAndDropSortInput')
+        state2.update_interaction_customization_args(
+            state2_customization_args_dict)
+        state2.update_interaction_answer_groups([state2_answer_group_dict])
+        state2.update_written_translations(
+            state_domain.WrittenTranslations.from_dict(
+                state_2_written_translations_dict))
+
+        exp_services.save_new_exploration('owner_id', exploration)
 
         html_list = (
-            exploration.init_state.interaction.get_all_html_content_strings())
+            init_state.get_all_html_content_strings() +
+            state2.get_all_html_content_strings() +
+            state1.get_all_html_content_strings())
         self.assertEqual(
             html_list,
             [
-                '<p>Feedback</p>', '<p>list_of_sets_of_html_strings</p>',
-                '<p>list_of_sets_of_html_strings</p>',
-                '<p>list_of_sets_of_html_strings</p>',
-                '<p>list_of_sets_of_html_strings</p>',
-                '<p>list_of_sets_of_html_strings</p>', '', ''])
+                '<p>init_state outcome html</p>',
+                '<p>Equals rule_spec html</p>',
+                '<p>ContainsAtLeastOneOf rule_spec html</p>',
+                '<p>IsProperSubsetOf rule_spec html</p>',
+                '<p>DoesNotContainAtLeastOneOf rule_spec html</p>',
+                '',
+                '<p>init_state customization arg html 1</p>',
+                '<p>init_state customization arg html 2</p>',
+                '<p>init_state customization arg html 3</p>',
+                '<p>init_state customization arg html 4</p>',
+                '',
+                '<p>state2 written_translation content-hi</p>',
+                '<p>state2 written_translation content-en</p>',
+                '<p>state2 written_translation outcome-hi</p>',
+                '<p>state2 written_translation outcome-en</p>',
+                '<p>state2 written_translation feedback-hi</p>',
+                '<p>state2 written_translation feedback-en</p>',
+                '<p>State 2 Feedback</p>',
+                '<p>IsEqualToOrdering rule_spec htmls</p>',
+                '<p>HasElementXAtPositionY rule_spec html</p>',
+                '<p>y input for HasElementXAtPositionY rule_spec </p>',
+                '<p>x input for HasElementXAtPositionY rule_spec </p>',
+                ('<p>IsEqualToOrderingWithOneItemAtIncorrectPosition rule_s'
+                 'pec htmls</p>'), '',
+                '<p>state 2 customization arg html 1</p>',
+                '<p>state 2 customization arg html 2</p>',
+                '<p>state 2 customization arg html 3</p>',
+                '<p>state 2 customization arg html 4</p>',
+                '<p>state2 content html</p>',
+                '<p>state 1 outcome html</p>', '',
+                '<p>state 1 content html</p>'])
+
+        html_field_types_to_rule_specs_dict = json.loads(
```

Comment:
Please make each of these things a separate test.

Reply:
done